### PR TITLE
00604 d eet testcases migration

### PR DESCRIPTION
**Pull request by ljianghedera**
**Related issue(s)**:
Address #604 and some issues opened before for this purpose.
 
**Summary of the change**:

1. Enhanced framework to allow transactions to be modified at last moment before submit to server;
2. Added test scenarios to EET suites to cover those tests currently exists only in legacy test clients.

Still two scenarios related to INVALID_SIGNATURE_COUNT_MISMATCHING_KEY and KEY_PREFIX_MISMATCH need more work.

**External impacts**:
None.

**Applicable documentation**
- [ ] Javadoc
- [ ] HAPI protobuf comments
- [ ] README

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/SystemDeleteContractSuite.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getContractInfo;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.fileCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.systemContractDelete;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.contractCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.systemContractUndelete;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.NOT_SUPPORTED;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+
+public class SystemDeleteContractSuite extends HapiApiSuite {
```
Comment (anighanta) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/SystemDeleteContractSuite.java:23]: I think you can add these tests to contractDeleteSuite instead of creating a new test class.

File: test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnUtils.java [hunk 2]
```diff
@@ -417,4 +421,72 @@ public static FeeData defaultPartitioning(FeeComponents components, int numPayer
...
+			txnBody.setMemo(newMemo);
+			return txn.toBuilder().setBodyBytes(txnBody.build().toByteString()).build();
+		} catch (Exception e) {
+			log.warn("Transaction's body can't be parsed: {}", txnToString(txn), e);
+		}
+		return null;
+	}
+
+	public static Transaction replaceTxnPayerAccount(Transaction txn, AccountID accountID) {
+		//log.info(String.format("Old Txn attr: %s", TxnUtils.txnToString(txn) ));
```
Comment (anighanta) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnUtils.java:439]: Please remove commented out code.

**Review by anighanta — DISMISSED** [review]

File: test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnUtils.java (hunk 2 above)
Comment (ljianghedera) [reply test-clients/src/main/java/com/hedera/services/bdd/spec/transactions/TxnUtils.java:439]: Done.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/SystemDeleteContractSuite.java (hunk 1 above)
Comment (ljianghedera) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/SystemDeleteContractSuite.java:23]: This is a good question. I've been thinking how to organize the test suites for some time. To me the big challenge of the test suite organization is to have biggest coverage with shortest test running time. To achieve this, the key is to reduce the redundancy. This will naturally lead people to put all test cases in one suite. This of course is unacceptable. So how to dice and slice the test cases is a good topic we need to discuss.
 
My current tentative thought is to put all ordinary small contract operations (Create/Read(query)/Update/Delete)  that are used by normal users day in day out into one suite, instead of separated by CRUD operations.  This one suite goes into CI workflow.  In addition to this, we will have several suites to cover more advanced, special corner cases  Any system account involved test cases should go into one or more suites of the advanced coverage suites. And these test suites will go to nightly regression workflow. 

File, crypto, consensus, token and any future services can follow same organization.

One issue with this is that it's not easy for people to know what test cases we already have. If a developer plans to add a new coverage, he or she may need to go through quite some test suites to find out if the test cases already exist.

I guess this will be the first step to move towards this direction.

File: test-clients/src/main/java/com/hedera/services/bdd/suites/file/negative/QueryFailuresSpec.java [hunk 3]
```diff
@@ -67,11 +68,11 @@ private HapiApiSpec getsExpectedRejections() {
 								.hasAnswerOnlyPrecheck(INVALID_FILE_ID),
 						getFileContents("tbd")
 								.nodePayment(1_234L)
-								.hasAnswerOnlyPrecheck(FILE_DELETED)
+								.hasAnswerOnlyPrecheck(OK)
```
Comment (anighanta) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/file/negative/QueryFailuresSpec.java:71]: Are we sending a response code `OK` even when the file is deleted ?

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/OCTokenSpec.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+						getAccountInfo("Bob").saveToRegistry("BobAcctInfo"),
+						getAccountInfo("Carol").saveToRegistry("CarolAcctInfo"),
+						getAccountInfo("Dave").saveToRegistry("DaveAcctInfo"),
+
+						fileCreate("bytecode")
+								.path(PATH_TO_OC_TOKEN_BYTECODE),
+
+						contractCreate("tokenContract", TOKEN_ERC20_CONSTRUCTOR_ABI,
+								1_000_000L, "OpenCrowd Token", "OCT")
+								.gas(250_000L)
```
Comment (anighanta) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/OCTokenSpec.java:87]: How did you get through the `GAS_LIMIT_EXCEEDED` here ?

Comment (ljianghedera) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/OCTokenSpec.java:85]: The root cause is in the HapiContractCreate class. The encoding needs to be encodeArguments(...) instead of encode(...).

File: test-clients/src/main/java/com/hedera/services/bdd/suites/file/negative/QueryFailuresSpec.java (hunk 3 above)
Comment (ljianghedera) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/file/negative/QueryFailuresSpec.java:71]: Yeah, that's what it does as of now. I think we need to look into it more.

Comment (anighanta) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/file/negative/QueryFailuresSpec.java:71]: Should we create an issue to track in future for this ?

Comment (ljianghedera) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/file/negative/QueryFailuresSpec.java:71]: Yeah, I will create one or more follow up issues to track this and other issues that may show up after merging into master.

**Review by anighanta — APPROVED** [review]
LGTM
